crates/bevy_time/src/time.rs

```diff
@@ -92,57 +123,190 @@ impl Time {
     /// }
     /// ```
     pub fn update_with_instant(&mut self, instant: Instant) {
-        if let Some(last_update) = self.last_update {
-            self.delta = instant - last_update;
-            self.delta_seconds_f64 = self.delta.as_secs_f64();
+        let raw_delta = instant - self.last_update.unwrap_or(self.startup);
+
+        let delta = if self.paused {
+            Duration::ZERO
+        } else if self.relative_speed != 1.0 {
+            // Avoid rounding errors when the relative speed is 1.
+            raw_delta.mul_f64(self.relative_speed)
+        } else {
+            raw_delta
+        };
+
+        if self.last_update.is_some() {
+            self.delta = delta;
             self.delta_seconds = self.delta.as_secs_f32();
+            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.raw_delta = raw_delta;
+            self.raw_delta_seconds = self.raw_delta.as_secs_f32();
+            self.raw_delta_seconds_f64 = self.raw_delta.as_secs_f64();
+        } else {
+            self.first_update = Some(instant);
         }
 
-        self.time_since_startup = instant - self.startup;
-        self.seconds_since_startup = self.time_since_startup.as_secs_f64();
+        self.elapsed_since_startup += delta;
+        self.seconds_since_startup = self.elapsed_since_startup.as_secs_f32();
+        self.seconds_since_startup_f64 = self.elapsed_since_startup.as_secs_f64();
+        self.raw_elapsed_since_startup += raw_delta;
+        self.raw_seconds_since_startup = self.raw_elapsed_since_startup.as_secs_f32();
+        self.raw_seconds_since_startup_f64 = self.raw_elapsed_since_startup.as_secs_f64();
```

Comment:
Same question as for `delta` : why not store only the `Duration` and call `as_secs_f*` in getters ?

Reply:
Same answer as the above.